tests/frame_manager_test.go

```diff
@@ -33,37 +32,106 @@ func TestWaitForFrameNavigationWithinDocument(t *testing.T) {
 	for _, tc := range testCases {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
-			tb := newTestBrowser(t, withFileServer())
-			p := tb.NewPage(nil)
-
-			resp := p.Goto(tb.staticURL("/nav_in_doc.html"), nil)
-			require.NotNil(t, resp)
-
-			// A click right away could possibly trigger navigation before we
-			// had a chance to call WaitForNavigation below, so give it some
-			// time to simulate the JS overhead, waiting for XHR response, etc.
-			time.AfterFunc(timeout*time.Millisecond, func() { //nolint:durationcheck
-				_ = tb.await(func() error {
-					_ = p.Click(tc.selector, nil)
-					return nil
-				})
-			})
+			t.Parallel()
 
-			done := make(chan struct{}, 1)
+			errc := make(chan error)
 			go func() {
-				require.NotPanics(t, func() {
-					p.WaitForNavigation(tb.toGojaValue(&common.FrameWaitForNavigationOptions{
-						Timeout: timeout * 3, // interpreted as ms
+				tb := newTestBrowser(t, withFileServer())
+				p := tb.NewPage(nil)
+
+				resp := p.Goto(tb.staticURL("/nav_in_doc.html"), tb.toGojaValue(&common.FrameGotoOptions{
+					WaitUntil: common.LifecycleEventNetworkIdle,
+					Timeout:   time.Duration(int64(timeout)) * time.Millisecond,
```

Comment:
This should just be:
```suggestion
					Timeout:   timeout,
```

Again, because of the way we parse the value, it's not intuitive using it from Go.

You can confirm this if you add `fmt.Printf("timeout: %s\n", parsedOpts.Timeout)` in `FrameManager.NavigateFrame()`. Currently it prints `got timeout: 1388h53m20s`, when it should be `5s`.

Reply:
Thanks, that was a bad copy paste. I've reverted the change and rebased it into the last commit